libcudacxx/include/nv/detail/__target_macros

```diff
@@ -347,6 +347,15 @@
 #    define _NV_TARGET_BOOL___NV_IS_EXACTLY_SM_90 0
 #  endif
 
+// SM90a support
+#  define NV_HAS_FEATURE_SM_90a   __NV_HAS_FEATURE_SM_90a
```

Comment:
The macro `NV_HAS_FEATURE_SM_90a` is only defined in the `_NV_COMPILER_NVCC` section.  That macro is supposed to be available for use in portable code, so it must always be defined.  It will never be true for any compiler other than NVCC (for now, at least), but it still must exist.

Reply:
Good point. I though it would be straight-forward to add a macro that is always false on nvc++, but I can't get past the `error: the condition of an "if target" must be a constexpr target architecture expression` error.

Current code:
```
#include <nv/target>

// Define for all compilers:
#    define NV_HAS_FEATURE_SM_90a   __NV_HAS_FEATURE_SM_90a

#if defined(_NV_COMPILER_NVCXX) && defined(_NVHPC_CUDA)
// nvc++
#     define _NV_TARGET___NV_HAS_FEATURE_SM_90a (0==1)
#else
// nvcc:
#  if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 900) && defined(__CUDA_ARCH_FEAT_SM90_ALL))
#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_90a 1
#  else
#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_90a 0
#  endif
#endif

// Type your code here, or load an example.
__global__ void kernel(int *out) {
    NV_IF_TARGET(
        NV_HAS_FEATURE_SM_90a, (
        *out = 901;
    ));
}
```
(compiler explorer for nvc++ link): https://ce.nvidia.com/z/aeMchn

How would I fix this?

Reply:
I don't claim that this is the best way to do it.  But this is the first thing that came to mind.

In the "all compilers" section
```
#define NV_HAS_FEATURE_SM_90a NV_NO_TARGET
```
then in the NVCC section, have the code that you added as it is, including defining `NV_HAS_FEATURE_SM_90a`, except with a
```
#undef NV_HAS_FEATURE_SM_90a
```
at the beginning.

Reply:
Wonderful! Thanks for the hint. I have pushed code that should do this now. Re-implementation in compiler explorer: https://ce.nvidia.com/z/zceq6K